source/docs/software/basic-programming/java-units.rst

```diff
@@ -1,58 +1,135 @@
+
+For local variables, you may choose to use Java's `var` keyword instead of including the full type name. For example, these are equivalent:
+
+.. code-block:: java
+
+  Measure<Per<Voltage, Velocity<Distance>>> v = VoltsPerMeterPerSecond.of(8);
+  var v = VoltsPerMeterPerSecond.of(8);
 
-New ``Measure`` objects can be created by calling the ``Unit#of`` method on the appropriate unit object, such as ``Inches.of(8)`` or ``Volts.of(13.5)``.
-``Measure`` is a generic type that contains the type of unit the measurement uses, such as ``Angle`` or ``Distance``. For example, ``Measure<Distance> wheelDiameter = Inches.of(6)`` is a measurement object corresponding to a distance of six inches, and ``Measure<Velocity<Angle>> maxRPM = RPM.of(5640)`` is a measurement corresponding to an angular velocity of 5,640 RPM.
+Creating Measures
+^^^^^^^^^^^^^^^^^
+
+The ``Measure`` class is a generic type that represents a magnitude (physical quantity) with its corresponding unit. It provides a consistent and type-safe way to handle different dimensions of measurements, such as distance, angle, and velocity, but abstracts away the particular unit (e.g. meter vs. inch). To create a ``Measure`` object, you call the ``Unit.of`` method on the appropriate unit object. For example, to create a ``Measure<Distance>`` object representing a distance of 6 inches, you would write:
 
 .. code-block:: java
 
-   import edu.wpi.first.units.Distance;
-   import edu.wpi.first.units.Measure;
-   import static edu.wpi.first.units.Units.*;
+  Measure<Distance> wheelDiameter = Inches.of(6);
 
-   public class Constants {
-     public static class DriveConstants {
-       /** The distance between the front and rear axles. **/
-       public static final Measure<Distance> kWheelBase = Inches.of(28);
+Other measures can also be created using their ``Unit.of`` method:
 
-       /** The distance between left and right wheels. **/
-       public static final Measure<Distance> kTrackWidth = Inches.of(26);
-     }
-   }
+.. code-block:: java
+
+  Measure<Mass> kArmMass = Kilograms.of(1.423);
+  Measure<Distance> kArmLength = Inches.of(32.25);
+  Measure<Angle> kMinArmAngle = Degrees.of(5);
+  Measure<Angle> kArmMaxTravel = Rotations.of(0.45);
+  Measure<Velocity<Distance> kMaxSpeed = MetersPerSecond.of(2.5);
+
+Performing Calculations
+^^^^^^^^^^^^^^^^^^^^^^^
 
-Because the ``Measure`` objects are Java objects, arithmetic must be done by calling methods on the objects. Note that each operation will return a new ``Measure`` object.
+The ``Measure`` class also supports arithmetic operations, such as addition, subtraction, multiplication, and division. These are done by calling methods on the objects. These operations always ensure that the units are compatible before performing the calculation, and they return a new ``Measure`` object. For example, you can add two ``Measure<Distance>`` objects together, even if they have different units:
```

Comment:
This section would be a good place to link to the GC article, since the default immutable operations could be a source of unexpected object allocations in hot loops. Or link to the mutability & object creation section below